lib/metadata_json_lint.rb

```diff
@@ -167,8 +167,8 @@ def validate_puppet_ver!(requirement)
     end
 
     return unless requirement.mixed_syntax?
-    warn(:dependencies, 'Mixing "x" or "*" version syntax with operators is not recommended in ' \
-      "metadata.json, use one style in the #{dep['name']} dependency: #{dep['version_requirement']}")
+    warn(:requirements, 'Mixing "x" or "*" version syntax with operators is not recommended in ' \
+      "metadata.json, use one style in the #{requirements['name']} dependency: #{requirements['version_requirement']}")
```

Comment:
This only has the variable `requirement` in the local scope. The name isn't known, but we do know that this is the puppet requirement. Perhaps `VersionRequirement` needs a `.to_s` for this.